web/app/Helpers/PageBuilder.php

```diff
@@ -0,0 +1,344 @@
+
+            /** We'll break the GET string into key-value pairs. */
+            $get_pairs = explode('&', $get_string);
+
+            /** Then break the key-value pairs into their respective items. */
+            foreach ($get_pairs as $get_pair) {
+                $item = explode('=', $get_pair);
+                $key = $item[0];
+                $value = $item[1];
+
+                /** We put the set in the same parameter store we're already building. */
+                $this->parameterArray[$key] = urldecode($value);
+                $this->parameterTypes[$key] = 'GET';
+                $this->allParameters['GET'][$key] = urldecode($value);
+            }
+
+            /** Lastly, POST values don't need as much manipulation. */
+            foreach ($_POST as $key => $value) {
+                /** Convert any CRLF values to LF */
+                $value = str_replace(["\r\n", "\r", "\n"], "\n", $value);
+
+                /** And add them to the bucket. */
+                $this->parameterArray[$key] = $value;
+                $this->parameterTypes[$key] = 'POST';
+                $this->allParameters['POST'][$key] = urldecode($value);
+            }
+        }
+    }
+
+    /**
+     *  Validate the requested template or screen and retrieve the location of it.
+     */
+    public function setTemplate()
+    {
+        /** The ajaxMode property can be set true by the AjaxModuleWebFlowController */
+        if ($this->ajaxMode === true) {
+            /** We can reliably assume that an AJAX module provided a key called moduleName */
+            $template = $this->parameterArray['moduleName'];
+
+            /**
+             * If we didn't get a valid value back (not empty, contains only
+             *  letters, numbers, underscores, and slashes), use the Empty template. (nothing)
+             */
+            if ($template == null || preg_match('/^[a-z0-9_\/]+$/i', $template) != 1) {
+                $template = 'Empty';
+            }
+
+            /**
+             * Otherwise we're going to assume there's a matching template in
+             *  templates/modules to go with the PHP module in php/modules.
+             *
+             * These AJAX-centric modules generate the HTML to be injected back
+             *  into the DOM wholesale.
+             */
+            $this->moduleTemplate = $template;
+
+            $this->findModuleClass($this->moduleTemplate);
+        } else {
+            /**
+             * If we're not making an AJAX call, we're looking for a Screen instead of a template.
+             * Screens are in php/Screens and are the classes that use the
+             * templates in templates/screens.
+             */
+            $template = $this->parameterArray['template'];
+
+            /**
+             * If we didn't get a valid value back (not empty, contains only
+             *  letters, numbers, underscores, and slashes), use the Index screen. (nothing)
```

Comment:
The `Index` screen is "nothing"?

Reply:
Indeed. It's a pointer to a file that doesn't exist. It should be at `web/templates/screens/IndexScreen.tpl` but the file isn't there at all.

https://github.com/scpwiki/wikijump/blob/WJ-771-login-and-registration/web/lib/ozoneframework/php/core/RunData.php#L256

The equivalent module template is `Empty`:

https://github.com/scpwiki/wikijump/blob/WJ-771-login-and-registration/web/templates/modules/Empty.tpl